src/ngraph/runtime/gpu/gpu_emitter.cpp

```diff
@@ -569,9 +569,65 @@ cudnnSetOpTensorDescriptor(opTensorDesc,
                 // Other cases (reordering of axes for tensors with rank>2) are not handled yet.
                 else
                 {
-                    throw runtime_error(
-                        "Axis permutation in reshape is not implemented yet for tensors with "
-                        "rank>2");
+                    size_t* input_strides = new size_t[arg_rank];
```

Comment:
Use a vector<size_t> here. Slightly less efficient but it does bounds checking.

Reply:
Yes, right, changed.